### PR TITLE
Chapter12

### PR DESCRIPTION
**Pull request by AndreeaSburlea**
## New features
- swipe for action left and right

## Changes in code
- delete action, share action and add to favorite action

## Screenshot

Screenshot 2022-04-26 at 11 31 06
Screenshot 2022-04-26 at 11 31 39
Screenshot 2022-04-26 at 11 31 28
Screenshot 2022-04-26 at 11 39 13
Screenshot 2022-04-26 at 11 39 41

### CODE REVIEW

**Review by madalinaCojocaru1 — APPROVED** [review]

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 1]
```diff
@@ -110,4 +107,82 @@ class RestaurantTableViewController: UITableViewController {
         // Deselect the row
         tableView.deselectRow(at: indexPath, animated: false)
     }
+
+    // Swipe left action
+    override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        // Get the selected restaurant
+        guard let restaurant = self.dataSource.itemIdentifier(for: indexPath)
+        else {
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: usually else is on the same line with the guard statement

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 2]
```diff
@@ -110,4 +107,82 @@ class RestaurantTableViewController: UITableViewController {
...
+        // Share action
+        let shareAction = UIContextualAction(style: .normal, title: "Share") {(_, _, completionHandler) in
+            let defaultText = "Just checking in at" + restaurant.name
+            let activityController: UIActivityViewController
+
+            if let imageToShare = UIImage(named: restaurant.image) {
+                activityController = UIActivityViewController(activityItems: [defaultText, imageToShare], applicationActivities: nil)
+            } else {
+                    activityController = UIActivityViewController(activityItems: [defaultText], applicationActivities: nil)
+                }
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: These lines are not aligned properly

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 3]
```diff
@@ -110,4 +107,82 @@ class RestaurantTableViewController: UITableViewController {
...
+
+        // Configure both actions as swipe action
+        let swipeConfiguration = UISwipeActionsConfiguration(actions: [deleteAction, shareAction])
+        return swipeConfiguration
+    }
+
+    // Swipe right action
+    override func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        // Get the selected restaurant
+        let _: UIActivityViewController
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: what is this used for?

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 4]
```diff
@@ -110,4 +107,82 @@ class RestaurantTableViewController: UITableViewController {
...
+        guard self.dataSource.itemIdentifier(for: indexPath) != nil
+        else {
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: same guard else issue

**Review by ralumarusca — CHANGES_REQUESTED** [review]

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift (hunk 3 above)
Comment (AndreeaSburlea) [reply FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: I used these to add a check-in action when we swipe right the cell. I add the swipe to favorite action.

**Review by ralumarusca — APPROVED** [review]

**Pull request merged**